From f7bcd2263ffbb2b62e2ac5d5c446d9cd5abdd3a2 Mon Sep 17 00:00:00 2001
From: theminer3746 <bwsminecraft@hotmail.com>
Date: Thu, 15 Dec 2016 22:07:16 +0700
Subject: [PATCH] Make change to how we handle exception

---
 app/Exceptions/Handler.php | 15 ++++++++++-----
 1 file changed, 10 insertions(+), 5 deletions(-)

diff --git a/app/Exceptions/Handler.php b/app/Exceptions/Handler.php
index 396757d..2f290b3 100644
--- a/app/Exceptions/Handler.php
+++ b/app/Exceptions/Handler.php
@@ -21,7 +21,7 @@ class Handler extends ExceptionHandler
         \Illuminate\Database\Eloquent\ModelNotFoundException::class,
         \Illuminate\Session\TokenMismatchException::class,
         \Illuminate\Validation\ValidationException::class,
-        Symfony\Component\HttpKernel\Exception\NotFoundHttpException::class,
+        \Symfony\Component\HttpKernel\Exception\NotFoundHttpException::class,
     ];
 
     /**
@@ -34,11 +34,16 @@ class Handler extends ExceptionHandler
      */
     public function report(Exception $exception)
     {
-        if($exception instanceof TokenMismatchException){
-            return redirect('/login');
+        if($exception instanceof \Illuminate\Session\TokenMismatchException){
+            $request->session()->flush();
+            $request->session()->regenerate();
+            return redirect('/supervisor_login')->with('error', 'การเชื่อมต่อหมดอายุ กรุณาเข้าสู่ระบบอีกครั้ง');
+        }
+
+        if(parent::shouldReport($exception)){
+            Log::error($exception);
+            parent::report($exception);
         }
-        Log::error($exception);
-        parent::report($exception);
     }
 
     /**
-- 
1.9.5.msysgit.0

